### PR TITLE
Added other valid values for backup protocol flag

### PR DESCRIPTION
**Pull request by phughk**
Added example values for online backup client protocol flag

### CODE REVIEW

**Review by martinfurmanski — APPROVED** [review]

File: enterprise/backup/src/main/java/org/neo4j/backup/impl/OnlineBackupContextBuilder.java [hunk 1]
```diff
@@ -117,15 +119,17 @@
 
     public static Arguments arguments()
     {
+        String argExampleProtoOverride =
+                String.join( "|", Stream.of( SelectedBackupProtocol.values() ).map( SelectedBackupProtocol::getName ).sorted().collect( Collectors.toList() ) );
```
Comment (praveenag) [inline enterprise/backup/src/main/java/org/neo4j/backup/impl/OnlineBackupContextBuilder.java]: Could use [Collectors#joining](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#joining--) instead.

File: enterprise/backup/src/main/java/org/neo4j/backup/impl/SelectedBackupProtocol.java [hunk 2]
```diff
@@ -24,7 +24,7 @@
 public enum SelectedBackupProtocol
 {
     ANY( "any" ),
-    LEGACY( "legacy" ),
+    LEGACY( "common" ),
```
Comment (praveenag) [inline enterprise/backup/src/main/java/org/neo4j/backup/impl/SelectedBackupProtocol.java]: If we are calling it `common` why not make the enum to `Common` instead.

File: enterprise/backup/src/test/java/org/neo4j/backup/impl/BackupHelpOutput.java [hunk 3]
```diff
@@ -67,7 +67,7 @@
         lines.add( "                                           backup will be attempted." );
         lines.add( "  --from=<address>                         Host and port of Neo4j." );
         lines.add( "                                           [default:localhost:6362]" );
-        lines.add( "  --protocol=<catchup>                     Preferred protocol to use for" );
+        lines.add( "  --protocol=<any|catchup|common>          Preferred protocol to use for" );
```
Comment (praveenag) [inline enterprise/backup/src/test/java/org/neo4j/backup/impl/BackupHelpOutput.java]: Could this instead say `Preferred backup protocol`

File: enterprise/backup/src/test/java/org/neo4j/backup/impl/OnlineBackupCommandCcIT.java [hunk 4]
```diff
@@ -205,7 +205,7 @@ public void backupNotPossibleIfLegacyProtocolSpecified() throws Exception
                         "--cc-report-dir=" + backupDir,
                         "--backup-dir=" + backupDir,
                         "--name=defaultport",
-                        "--proto=legacy") );
+                        "--proto=common") );
```
Comment (praveenag) [inline enterprise/backup/src/test/java/org/neo4j/backup/impl/OnlineBackupCommandCcIT.java]: Shouldn't this say `--protocol` instead of `--proto`

**Review by praveenag — CHANGES_REQUESTED** [review]

**Review by praveenag — APPROVED** [review]

**Pull request merged**